From: Pino Toscano <toscano.pino@tiscali.it>
Date: Sun, 16 Jul 2023 11:34:53 +0200
Subject: Fallback on _POSIX_PIPE_BUF w/ missing PIPE_BUF

Forwarded: https://github.com/janestreet/spawn/pull/49
Last-Update: 2023-07-03

PIPE_BUF is optional in POSIX, e.g. "where the corresponding value is
equal to or greater than the stated minimum, but where the value can
vary depending on the file to which it is applied." [1]

GNU/Hurd does not provide PIPE_BUF, so fallback to its minimum
acceptable value, that is _POSIX_PIPE_BUF.

[1] https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/limits.h.html
---
 vendor/spawn/src/spawn_stubs.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/vendor/spawn/src/spawn_stubs.c b/vendor/spawn/src/spawn_stubs.c
index 870baa0..29c7eb3 100644
--- a/vendor/spawn/src/spawn_stubs.c
+++ b/vendor/spawn/src/spawn_stubs.c
@@ -164,7 +164,11 @@ static void subprocess_failure(int failure_fd,
   sigset_t sigset;
   ssize_t written;
 
+#ifdef PIPE_BUF
   CASSERT(sizeof(failure) < PIPE_BUF)
+#else
+  CASSERT(sizeof(failure) < _POSIX_PIPE_BUF)
+#endif
 
   set_error(&failure, errno, function, error_arg);
 
